### PR TITLE
Add deepcopy generator checker

### PR DESCRIPTION
**Pull request by aanm**
To avoid code from being merged without being regenerated, this commit adds a checker, that will only run in the CI, to verify if there is code ungenerated.

### CODE REVIEW

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

File: contrib/scripts/check-k8s-code-gen.sh [hunk 1]
```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env bash
+
+set -e
+set -o pipefail
+
+# Delete all zz_generated.deepcopy.go files
+find . -not -path "./vendor/*" -iname *"zz_generated.deepcopy.go" -exec rm  {} \;
```
Comment (christarazi) [inline contrib/scripts/check-k8s-code-gen.sh]:
```suggestion
find . -not -path "./vendor/*" -iname "*zz_generated.deepcopy.go" -exec rm  {} \;
```

Did you mean this instead?

**Review by christarazi — APPROVED** [review]
Looks good beside one potential typo

File: contrib/scripts/check-k8s-code-gen.sh (hunk 1 above)
Comment (aanm) [reply contrib/scripts/check-k8s-code-gen.sh]: yes :facepalm:

**Comment by aanm** [comment]
test-me-please

**Review by qmonnet — APPROVED** [review]
Looks good, the checker will be great.

Does any of the fix commits need a `fixes:` tag? I think the code missing deepcopy was added recently and no backport is necessary here, right? (Just double-checking)

**Comment by aanm** [comment]
@qmonnet that is correct. I didn't had a fixes because it because of multiple commits and this is always happening.

**Comment by qmonnet** [comment]
test-gke

**Comment by aanm** [comment]
@qmonnet GKE is not marked as "Required" for a PR to be merged.

**Comment by aanm** [comment]
`Cilium-Tests-With-Kernel` was renamed to `K8s-1.17-Kernel-4.19`

**Review by nebril — APPROVED** [review]

**Pull request merged**